modules/base/rendering/renderablenodedirectionhint.cpp

```diff
@@ -0,0 +1,544 @@
+        std::optional<bool> useRelativeLength;
+
+        // [[codegen::verbatim(WidthInfo.description)]]
+        std::optional<float> width [[codegen::greaterequal(0.f)]];
+
+        // [[codegen::verbatim(ShadingEnabledInfo.description)]]
+        std::optional<float> performShading;
+
+        // [[codegen::verbatim(AmbientIntensityInfo.description)]]
+        std::optional<float> ambientIntensity [[codegen::greaterequal(0.f)]];
+
+        // [[codegen::verbatim(DiffuseIntensityInfo.description)]]
+        std::optional<float> diffuseIntensity [[codegen::greaterequal(0.f)]];
+
+        // [[codegen::verbatim(SpecularIntensityInfo.description)]]
+        std::optional<float> specularIntensity [[codegen::greaterequal(0.f)]];
+    };
+#include "renderablenodedirectionhint_codegen.cpp"
+} // namespace
+
+namespace openspace {
+
+documentation::Documentation RenderableNodeDirectionHint::Documentation() {
+    return codegen::doc<Parameters>("base_renderable_renderablenodedirectionhint");
+}
+
+RenderableNodeDirectionHint::Shading::Shading()
+    : properties::PropertyOwner({ "Shading" })
+    , enabled(ShadingEnabledInfo, true)
+    , ambientIntensity(AmbientIntensityInfo, 0.2f, 0.f, 1.f)
+    , diffuseIntensity(DiffuseIntensityInfo, 0.7f, 0.f, 1.f)
+    , specularIntensity(SpecularIntensityInfo, 0.f, 0.f, 1.f)
+{
+    addProperty(enabled);
+    addProperty(ambientIntensity);
+    addProperty(diffuseIntensity);
+    addProperty(specularIntensity);
+}
+
+RenderableNodeDirectionHint::RenderableNodeDirectionHint(const ghoul::Dictionary& dictionary)
+    : Renderable(dictionary)
+    , _start(StartNodeInfo, "Root")
+    , _end(EndNodeInfo, "Root")
+    , _color(ColorInfo, glm::vec3(1.f), glm::vec3(0.f), glm::vec3(1.f))
+    , _segments(SegmentsInfo, 10, 3, 100)
+    , _invertArrowDirection(InvertInfo, false)
+    , _arrowHeadSize(ArrowHeadSizeInfo, 0.1f, 0.f, 1.f)
+    , _arrowHeadWidthFactor(ArrowHeadWidthInfo, 2.f, 1.f, 100.f)
+    , _offsetDistance(OffsetDistanceInfo, 0.f, 0.f, 1e20f)
+    , _useRelativeOffset(RelativeOffsetInfo, false)
+    , _length(LengthInfo, 100.f, 0.f, 1e20f)
+    , _useRelativeLength(RelativeLengthInfo, false)
+    , _width(WidthInfo, 10.f, 0.f, 1e11f)
+{
+    const Parameters p = codegen::bake<Parameters>(dictionary);
+
+    addPropertySubOwner(_shading);
+    _shading.enabled = p.performShading.value_or(_shading.enabled);
+    _shading.ambientIntensity = p.ambientIntensity.value_or(_shading.ambientIntensity);
+    _shading.diffuseIntensity = p.diffuseIntensity.value_or(_shading.diffuseIntensity);
+    _shading.specularIntensity = p.specularIntensity.value_or(_shading.specularIntensity);
+
+    _start = p.startNode.value_or(_start);
+    addProperty(_start);
+
+    _end = p.endNode.value_or(_end);
+    addProperty(_end);
+
+    _color = p.color.value_or(_color);
+    _color.setViewOption(properties::Property::ViewOptions::Color);
+    addProperty(_color);
+
+    addProperty(_opacity);
```

Comment:
```suggestion
    addProperty(Fadeable::_opacity);
```
to make it more clear where it comes from